src/OpenTK/Platform/Windows/WinDisplayDevice.cs

```diff
@@ -32,18 +32,88 @@
 
 namespace OpenTK.Platform.Windows
 {
-    internal sealed class WinDisplayDeviceDriver : DisplayDeviceBase
+    internal sealed class WinDisplayDeviceDriver : IDisplayDeviceDriver
     {
         private readonly object display_lock = new object();
 
+        // Small object for tracking name, available resolutions, and original resolution.
+        private sealed class WinDevice
+        {
+            public int DevNum;
+            public string Name;
+            public DisplayResolution OriginalResolution;
+            public List<DisplayResolution> AvailableResolutions;
+        }
+
+        private DisplayDevice _primary;
+        private List<DisplayDevice> _availableDevices = new List<DisplayDevice>();
+
         public WinDisplayDeviceDriver()
         {
             RefreshDisplayDevices();
-            SystemEvents.DisplaySettingsChanged +=
-                HandleDisplaySettingsChanged;
+            SystemEvents.DisplaySettingsChanged += HandleDisplaySettingsChanged;
+        }
+
+        public DisplayDevice GetDisplay(DisplayIndex index)
```

Comment:
Yeh at one point it looked like they were going to end up being different, but ended up converging to the same code so I'll add the base class back in.